TribalWars/src/ApplicationCore/Interfaces/IUserService.cs

```diff
@@ -0,0 +1,15 @@
+using System.Threading.Tasks;
+using ApplicationCore.DTOs;
+using ApplicationCore.Results;
+
+namespace ApplicationCore.Interfaces
+{
+    public interface IUserService
+    {
+        Task<(ServiceResult, string)> CreateUserAsync(UserDto userDto, string password);
```

Comment:
Instead of using tuple, you could create a generic ServiceResult subclass like this:

```
public class ServiceResult<T> :  ServiceResult
    {
        public T Content { get; set; }
    }
```

then your method definition would look like this:

```
Task<ServiceResult<string>> CreateUserAsync....
```

It makes the intention behind return type a bit more readable and response easier to deal with in higher layers (single object).

Reply:
I was wondering to implement that approach, but I thought: what if I wanted to return ServiceResult without any data? I'd have to use e.g. bool as generic type, it isn't big deal. But next uncertainty: In both of those approaches you don't know what exactly will be returned in that string, but using tuple, you can do that in higher classes:
`var (result, emailVerifyToken) = await _userService.CreateUserAsync...`
instead of:
`var result = await _userService.CreateUserAsync...`
Isn't it more readable?